Aztec/Classes/Plugin/Plugin.swift

```diff
@@ -0,0 +1,46 @@
+import Foundation
+
+open class Plugin {
+    
+    public init() {}
+    
+    public func loaded(into textView: TextView) {}
+    
+    open func process(inputHTML html: String) -> String { return html }
+    open func process(outputHTML html: String) -> String { return html }
+
+    open func getInputHTMLTreeProcessor() -> HTMLTreeProcessor? { return nil }
+    open func getOutputHTMLTreeProcessor() -> HTMLTreeProcessor? { return nil }
+    open func getElementConverter() -> ElementConverter? { return nil }
+
+    public static func ==(lhs: Plugin, rhs: Plugin) -> Bool {
+        return type(of: lhs) == type(of: rhs)
+    }
+}
+
+extension Array where Element: Plugin {
```

Comment:
IMHO opinion I think this are better of has methods on the TextStorage class than extension to Array. 

Like the extension on NSAtributtedString that you removed above I don't see this being used in another places.